src/gui/src/InstrumentEditor/InstrumentEditor.cpp

```diff
@@ -572,7 +590,16 @@ void InstrumentEditor::selectedInstrumentChangedEvent()
 		m_pResonanceRotary->setValue( m_pInstrument->get_filter_resonance() );
 		//~ filter
 
-		// random pitch
+		// pitch offset + random
+		char tmp[20];
+		sprintf( tmp, "%#.2f", m_pInstrument->get_pitch_offset() );
+		m_pPitchLCD->setText( tmp );
+		
+		int nCoarsePitch = (int) ::round(m_pInstrument->get_pitch_offset());
```

Comment:
I must have copied that part from the instrument layer editor (which has pitch shift) I guess, cause I would have not used `::` before `round`...
It's cast maybe because `round()` returns a `double`?
Not sure to follow the question but coarse is` int` (number of half steps) while fine is `float` (fractions of half steps)

Reply:
Or you mean why defining a nCoarsePitch of type int to pass into m_pPitchCoarseRotary->setValue() if it expects a float? It looks confused
I am revising this part!

Reply:
Something like:
```
/* fCoarsePitch is the closest integer to pitch_offset (represents the pitch shift interval in half steps)
while it is an integer number, it's defined float to be used in next lines */
float fCoarsePitch = round( m_pInstrument->get_pitch_offset() );

//fFinePitch represents the fine adjustment (between -0.5 and +0.5) if pitch_offset has decimal part
float fFinePitch = m_pInstrument->get_pitch_offset() - fCoarsePitch;

m_pPitchCoarseRotary->setValue( fCoarsePitch );
m_pPitchFineRotary->setValue( fFinePitch * 100 );
```

Reply:
> It's cast maybe because round() returns a double?

[round](https://en.cppreference.com/w/cpp/numeric/math/round) is overloaded and can return a `float` as well.

> Not sure to follow the question but coarse is int (number of half steps) while fine is float (fractions of half steps).

Well, you are right. Coarse is an integer (a whole number). But there is nothing wrong with representing it as a float as long as it makes the code more consistent.  Also, using `round` instead of rounding it using a cast makes your intentions for other people reading the code much more explicit.

> Why static_cast<int>() is better than (int)?

It's a thing (the) C++ coding standard [advertises](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-casts-named). For such simple numeric types it's most often not that relevant but also let's you spot it more clearly right away. But you don't have to add it. It can already be regarded a matter of taste

> Or you mean why defining a nCoarsePitch of type int to pass into m_pPitchCoarseRotary->setValue() if it expects a float? It looks confused

Yes. This is what I referred to with consistency above.

Reply:
Ok